Cpp_test/lib/ipc-shm.cpp

```diff
@@ -0,0 +1,58 @@
+#include "ipc-shm.h"
+
+IPCShm::~IPCShm()
+{
+	munmap(this->shm_ptr, this->shm_size_in_bytes);
```

Comment:
You should avoid munmapping a nullptr (https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/11284), just check if it's a nullptr before munmaping. But it should be initialized as nullptr.